src/Razor/test/Microsoft.AspNetCore.Razor.Test.Common/UseExportProviderAttribute.cs

```diff
@@ -0,0 +1,235 @@
+        /// to completion. If this timeout is exceeded by the asynchronous operations running after a test completes,
+        /// the test is failed.
+        /// </summary>
+        private static readonly TimeSpan CleanupTimeout = TimeSpan.FromMinutes(1);
+
+#if false
+        private MefHostServices? _hostServices;
+#endif
+
+        public override void Before(MethodInfo? methodUnderTest)
+        {
+#if false
+            MefHostServices.TestAccessor.HookServiceCreation(CreateMefHostServices);
+
+            // make sure we enable this for all unit tests
+            AsynchronousOperationListenerProvider.Enable(enable: true, diagnostics: true);
+#endif
+
+            ExportProviderCache.SetEnabled_OnlyUseExportProviderAttributeCanCall(true);
+        }
+
+        /// <summary>
+        /// To the extent reasonably possible, this method resets the state of the test environment to the same state as
+        /// it started, ensuring that tests running in sequence cannot influence the outcome of later tests.
+        /// </summary>
+        /// <remarks>
+        /// <para>The test cleanup runs in two primary steps:</para>
+        /// <list type="number">
+        /// <item>Waiting for asynchronous operations started by the test to complete.</item>
+        /// <item>Disposing of mutable resources created by the test.</item>
+        /// <item>Clearing static state variables related to the use of MEF during a test.</item>
+        /// </list>
+        /// </remarks>
+        public override void After(MethodInfo? methodUnderTest)
+        {
+            try
+            {
+                foreach (var exportProvider in ExportProviderCache.ExportProvidersForCleanup)
+                {
+                    DisposeExportProvider(exportProvider);
+                }
+            }
+            finally
+            {
+#if false
+                // Replace hooks with ones that always throw exceptions. These hooks detect cases where code executing
+                // after the end of a test attempts to create an ExportProvider.
+                MefHostServices.TestAccessor.HookServiceCreation(DenyMefHostServicesCreationBetweenTests);
+#endif
+
+                // Reset static state variables.
+#if false
+                _hostServices = null;
+#endif
+                ExportProviderCache.SetEnabled_OnlyUseExportProviderAttributeCanCall(false);
+            }
+        }
+
+        private static void DisposeExportProvider(ExportProvider? exportProvider)
+        {
+            if (exportProvider == null)
+            {
+                return;
+            }
+
+            // Dispose of the export provider, including calling Dispose for any IDisposable services created during the test.
+            using var _ = exportProvider;
+
+#if false
+            if (exportProvider.GetExportedValues<IAsynchronousOperationListenerProvider>().SingleOrDefault() is { } listenerProvider)
```

Comment:
📝 This block is where we can hook into the test cleanup process to ensure the test waits for asynchronous operations to complete before moving on to the next test. We'd need to update production code to enable this functionality.